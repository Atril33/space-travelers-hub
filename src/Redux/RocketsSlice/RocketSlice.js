import { createSlice } from '@reduxjs/toolkit';
import data from '../../components/TempRocketData';

const initialState = {
  RocketDatile: data,
};

const rocketSlice = createSlice({
  name: 'rocket',
  initialState,
});

export default rocketSlice.reducer;
