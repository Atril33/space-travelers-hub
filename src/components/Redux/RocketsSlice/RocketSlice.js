import { createSlice } from '@reduxjs/toolkit';
import data from '../../TempRocketData';

const initialState = {
  RocketDatile: data,
};

const rocketSlice = createSlice({
  name: 'rocket',
  initialState,
});

export default rocketSlice.reducer;
