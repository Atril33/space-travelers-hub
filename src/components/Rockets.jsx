import { useSelector } from 'react-redux';
import './style.css';

const Rockets = () => {
  const { RocketDatile } = useSelector((store) => store.rocket);
  return (
    <div className="rocket-container">
      {RocketDatile.map((item) => (
        <div className="rocket-items" key={item.name}>
          <img src={item.image} alt="Rocket" />
          <ul className="rocket-detail">
            <li><h2 className="rocket-name">{item.name}</h2></li>
            <li><p className="rocket-des">{item.des}</p></li>
            <li><button type="button" className="reserve-bttn">reserve rocket</button></li>
          </ul>
        </div>
      ))}

    </div>
  );
};

export default Rockets;
