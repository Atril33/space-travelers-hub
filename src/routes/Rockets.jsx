import { useSelector } from 'react-redux';
import '../styles/style.css';

const Rockets = () => {
  const { RocketDatile, buttonText } = useSelector((store) => store.rocket);

  return (
    <div className="rocket-container">
      {RocketDatile.map((item) => (
        <div className="rocket-items" key={item.id}>
          <img src={item.image} alt="Rocket" />
          <ul className="rocket-detail">
            <li><h2 className="rocket-name">{item.name}</h2></li>
            <li><p className="rocket-des">{item.description}</p></li>
            <li><button type="button" className="reserve-bttn">{buttonText}</button></li>
          </ul>
        </div>
      ))}

    </div>
  );
};

export default Rockets;
