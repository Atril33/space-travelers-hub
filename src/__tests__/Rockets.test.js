import { render, fireEvent } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import Rockets from '../routes/Rockets';
import { reserveActive, cancelReserve } from '../Redux/RocketsSlice/RocketSlice';

jest.mock('react-redux');

describe('Rockets Component', () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    useSelector.mockImplementation((selector) => selector({
      rocket: {
        RocketDatile: [
          {
            name: 'rocket1',
            description: 'Rocket 1 Description',
            image: 'rocket1.jpg',
            reserved: false,
          },
          {
            name: 'rocket2',
            description: 'Rocket 2 Description',
            image: 'rocket2.jpg',
            reserved: true,
          },
        ],
      },
    }));
    useDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('Should render Rockets on the Screen', () => {
    const { getByText, getAllByAltText } = render(<Rockets />);
    const rocketNames = getAllByAltText(/rocket/).map((element) => element.getAttribute('alt'));
    expect(rocketNames).toEqual(['rocket1', 'rocket2']);
    expect(getByText('Rocket 1 Description')).toBeInTheDocument();
    expect(getByText('Rocket 2 Description')).toBeInTheDocument();
  });

  test('Should dispatch cancelReserve action when cancel reservation button is clicked', () => {
    const { getByText } = render(<Rockets />);
    const cancelButton = getByText('cancel Reservation');
    fireEvent.click(cancelButton);
    expect(mockDispatch).toHaveBeenCalledWith(cancelReserve('rocket2'));
  });

  test('Should dispatch reserveActive action when reserve rocket button is clicked', () => {
    const { getByText } = render(<Rockets />);
    const reserveButton = getByText('Reserve Rocket');
    fireEvent.click(reserveButton);
    expect(mockDispatch).toHaveBeenCalledWith(reserveActive('rocket1'));
  });
});
