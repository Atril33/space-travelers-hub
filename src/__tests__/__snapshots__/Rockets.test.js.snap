// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets should match the snapshot 1`] = `
<div>
  <div
    class="rocket-container"
  >
    <div
      class="rocket-items"
    >
      <img
        alt="Falcon 1"
        src="https://imgur.com/DaCfMsj.png"
      />
      <ul
        class="rocket-detail"
      >
        <li>
          <h2
            class="rocket-name"
          >
            Falcon 1
          </h2>
        </li>
        <li>
          <p
            class="rocket-des"
          >
            The Falcon 1 was the first rocket developed by SpaceX.
          </p>
        </li>
        <button
          class="reserve-bttn"
          type="button"
        >
          Reserve Rocket
        </button>
      </ul>
    </div>
    <div
      class="rocket-items"
    >
      <img
        alt="Falcon 9"
        src="https://imgur.com/4Y2G3Um.png"
      />
      <ul
        class="rocket-detail"
      >
        <li>
          <h2
            class="rocket-name"
          >
            Falcon 9
          </h2>
        </li>
        <li>
          <p
            class="rocket-des"
          >
            The Falcon 9 is a partially reusable two-stage-to-orbit medium-lift launch vehicle.
          </p>
        </li>
        <button
          class="reserve-bttn"
          type="button"
        >
          Reserve Rocket
        </button>
      </ul>
    </div>
    <div
      class="rocket-items"
    >
      <img
        alt="Falcon Heavy"
        src="https://imgur.com/0M3WgjB.png"
      />
      <ul
        class="rocket-detail"
      >
        <li>
          <h2
            class="rocket-name"
          >
            Falcon Heavy
          </h2>
        </li>
        <li>
          <span
            class="reserve-active"
          >
            Reserved
          </span>
          <p
            class="rocket-des"
          >
            The Falcon Heavy is currently the most powerful operational rocket in the world.
          </p>
        </li>
        <button
          class="cancle-reserve"
          type="button"
        >
          cancel Reservation
        </button>
      </ul>
    </div>
  </div>
</div>
`;
